### PR TITLE
Load balancing documentation first draft

### PR DESCRIPTION
**Pull request by dgquintas**

### CODE REVIEW

**Comment by dgquintas** [comment]
@iamqizhao @zhangkun83

File: doc/load-balancing.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+latency to the RPCs.
+
+The proxy model was deemed inefficient when considering request heavy services
+like storage. 
+
+### Balancing-aware Client
+
+This thicker client places more of the load balancing logic in the client. For
+example, the client could contain many load balancing policies (Round Robin,
+Random, etc) used to select servers from a list. In this model, a load balancer
```
Comment (zhangkun83) [inline doc/load-balancing.md]: There wouldn't necessarily be a load-balancer. The server list may be statically configured or directly from the name resolution system.

File: doc/load-balancing.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+the preferred server from the list. 
+
+One of the drawbacks of this approach is writing and maintaining the load
+balancing policies in multiple languages and/or versions of the clients. These
+policies can be fairly complicated. Some of the algorithms also require client
+to server communication so the client would need to get thicker to support
+additional RPCs to get health or load information in addition to sending RPCs
+for user requests.
+
+It would also significantly complicate the API: the new design hides the load
```
Comment (zhangkun83) [inline doc/load-balancing.md]: What API are you referring to?

File: doc/load-balancing.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+balancing complexity of multiple layers and presents it as a simple list of
+servers to the client.
+
+### External Load Balancing Service
+
+The client load balancing code is kept simple and portable, implementing
+straightforward algorithms (ie, Pick First, Round Robin) for server selection.
```
Comment (zhangkun83) [inline doc/load-balancing.md]: Pick-first is not a well-known algorithm. Should have it defined.

File: doc/load-balancing.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+## Load Balancer
+
+The Load Balancer is responsible for providing the client with a list of servers
+and client RPC parameters. The balancer chooses when to update the list of
+servers and can decide whether to provide a complete list, a subset, or a
+specific list of “picked” servers in a particular order. The balancer can
+optionally provide an expiration interval after which the server list should no
+longer be trusted and should be updated by the balancer.
+
+The load balancer is may open reporting streams to each server contained in the
```
Comment (zhangkun83) [inline doc/load-balancing.md]: Delete "is"

Comment (dgquintas) [inline doc/load-balancing.md]: Done

File: doc/load-balancing.md (hunk 3 above)
Comment (dgquintas) [inline doc/load-balancing.md]: Simply dropped, a single example suffices.

File: doc/load-balancing.md (hunk 2 above)
Comment (dgquintas) [inline doc/load-balancing.md]: Reworded

File: doc/load-balancing.md (hunk 1 above)
Comment (dgquintas) [inline doc/load-balancing.md]: Done.

**Comment by dgquintas** [comment]
Any further comments?

**Comment by dgquintas** [comment]
I'll merge/publish by EOD today if there aren't any further concerns.

**Pull request merged**